### PR TITLE
add combined compression encryption test

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/9978/

File: js/client/modules/@arangodb/process-utils.js [hunk 1]
```diff
@@ -125,6 +125,16 @@ class ConfigBuilder {
     this.config['maskings'] = fs.join(TOP_DIR, "tests/js/common/test-data/maskings", dir);
   }
   activateEncryption() { this.config['encryption.keyfile'] = fs.join(this.rootDir, 'secret-key'); }
+  activateCompression() {
+    if (this.type == 'dump') {
+      this.config['--compress-output'] = true;
+    }
+  }
+  activateUNCompression() {
```
Comment (danielhlarkin) [inline js/client/modules/@arangodb/process-utils.js:133]:
```suggestion
  deactivateCompression() {
```

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/9979/

File: js/client/modules/@arangodb/testsuites/dump.js [hunk 2]
```diff
@@ -100,6 +100,14 @@ class DumpRestoreHelper {
...
       this.restoreOldConfig.activateEncryption();
     }
+    if (options.compressed) {
+      this.dumpConfig.activateCompression();
+      this.restoreOldConfig.activateCompression();
+    }
+    if (options.uncompressed) {
+      this.dumpConfig.activateUNCompression();
+      this.restoreOldConfig.activateUNCompression();
+    }
```
Comment (danielhlarkin) [inline js/client/modules/@arangodb/testsuites/dump.js:110]:
```suggestion
    if (dumpOptions.compressed) {
      this.dumpConfig.activateCompression();
    }
    if (dumpOptions.uncompressed) {
      this.dumpConfig.deactivateCompression();
    }
```

File: js/client/modules/@arangodb/testsuites/dump.js [hunk 3]
```diff
@@ -161,10 +169,18 @@ class DumpRestoreHelper {
     return this.validate(this.results.setup);
   }
 
+  runCheckDumpFilesSuite(path) {
+    this.print('Inspecting dumped files');
+    process.env['dumpdirectory'] = this.dumpConfig.config['output-directory'];
```
Comment (danielhlarkin) [inline js/client/modules/@arangodb/testsuites/dump.js:174]:
```suggestion
    process.env['dump-directory'] = this.dumpConfig.config['output-directory'];
```

File: js/client/modules/@arangodb/testsuites/dump.js [hunk 4]
```diff
@@ -161,10 +169,18 @@ class DumpRestoreHelper {
...
+    this.results.checkDumpFiles = this.arangosh(path, this.clientAuth);
+    process.env['dumpdirectory'] = null;
```
Comment (danielhlarkin) [inline js/client/modules/@arangodb/testsuites/dump.js:176]:
```suggestion
    delete process.env['dump-directory'];
```

File: tests/js/server/dump/dump-check-dump-files-compressed.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+let jsunity = require('jsunity');
+let internal = require('internal');
+let arangodb = require('@arangodb');
+let fs = require('fs');
+let pu = require('@arangodb/process-utils');
+let db = arangodb.db;
+
+function dumpIntegrationSuite () {
+  'use strict';
+  const dumpDir = process.env['dumpdirectory'];
```
Comment (danielhlarkin) [inline tests/js/server/dump/dump-check-dump-files-compressed.js:38]:
```suggestion
  const dumpDir = process.env['dump-directory'];
```

File: tests/js/server/dump/dump-check-dump-files-encrypted.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+let jsunity = require('jsunity');
+let internal = require('internal');
+let arangodb = require('@arangodb');
+let fs = require('fs');
+let pu = require('@arangodb/process-utils');
+let db = arangodb.db;
+
+function dumpIntegrationSuite () {
+  'use strict';
+  const dumpDir = process.env['dumpdirectory'];
```
Comment (danielhlarkin) [inline tests/js/server/dump/dump-check-dump-files-encrypted.js:38]:
```suggestion
  const dumpDir = process.env['dump-directory'];
```

File: tests/js/server/dump/dump-check-dump-files-uncompressed.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+let internal = require('internal');
+let arangodb = require('@arangodb');
+let fs = require('fs');
+let pu = require('@arangodb/process-utils');
+let db = arangodb.db;
+
+function dumpIntegrationSuite () {
+  'use strict';
+  // this file is used by multiple, hence the checked structure is only in a subdirectory:
+  const dumpDir = fs.join(process.env['dumpdirectory'], 'UnitTestsDumpSrc');
```
Comment (danielhlarkin) [inline tests/js/server/dump/dump-check-dump-files-uncompressed.js:39]:
```suggestion
  const dumpDir = fs.join(process.env['dump-directory'], 'UnitTestsDumpSrc');
```

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/9982/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/9984/

**Comment by dothebart** [comment]
unrelated errors in testrun.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/9987/

**Comment by dothebart** [comment]
tests green

**Pull request merged**